Automattic/jetpack
# CLI: Add basic CLI and build command

This adds a basic CLI framework with an initial command: `jetpack build`.

At this point, it is written using JS modules, but I don't transpile it down to CommonJS. I'm including `esm` module and letting node run it directly. If we need to add a build step for the cli, we can do that, but opting not to add that complexity until it is needed.

My long-term goal is to replace or at least provide easier discoverability of the `yarn` scripts in `package.json` that actually matter to individual devs.

#### Note for development ####
To develop the CLI, you'll need to tell yarn to use the local files instead of moving them into `node_modules`. To do so, `cd tools/cli` and `yarn link`. Then cd back to the repo root and `yarn link jetpack-cli`.

#### Changes proposed in this Pull Request:
* Adds basic development CLI.
* Adds test runner that executes on `yarn test-cli` and adds it to the js-tests GH action.

#### Jetpack product discussion
None yet.

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
* Run `yarn install`. The CLI is a node module that is symlinked, but needs to be "installed". This also should make the `jetpack` bin work too.
* To run the unit tests: `./node_modules/.bin/mocha tools/cli/tests/** --require esm`
* Run `jetpack` -- it should provide usage information and indicate that you need to run a command.
* Run `jetpack build` -- it should let you indicate a project.
* Run `jetpack build plugins/jetpack` it should run the JP build process.
* Run `jetpack build plugins/jetpack --production` it should run a production JP build.
* Try `jetpack build` selecting a package other than lazy-images.
* Try `jetpack build` selecting the package `lazy-images` (which does have a build step).

#### Proposed changelog entry for your changes:
* None at this time.
